python/ray/tests/test_basic_5.py

```diff
@@ -0,0 +1,82 @@
+# coding: utf-8
+import gc
+import logging
+import os
+import sys
+import time
+
+import pytest
+
+import ray.cluster_utils
+from ray._private.test_utils import wait_for_pid_to_exit
+
+import ray
+
+logger = logging.getLogger(__name__)
+
+
+def test_background_tasks_with_max_calls(shutdown_only):
+    ray.init(
+        # TODO (Alex): We need to fix
+        # https://github.com/ray-project/ray/issues/20203 to remove this flag.
+        num_cpus=2,
+        _system_config={"worker_cap_initial_backoff_delay_ms": 0})
+
+    num_tasks = 3
```

Comment:
Introduced this variable to see if the tests now passes within the bounds of CI. The time taken to shutdown is also fine (like a few milliseconds as compared to the total test duration). So, the other problem might be with memory. May be reducing number of tasks can help in effectively using the limited CI resources.

Reply:
Seems to work. `test_basic_5` now passes in 27s on Github CI with reduced number of tasks (10 -> 3).

Reply:
Another thing to be done is check if it works with 16 GB RAM and 2 CPUs.

Reply:
It works with 2 CPUs and 16 GB RAM but takes a longer time ( > 90ish s). Similarly for 8 CPUs and 4 GB RAM it takes a longer time (~ 60ish s). And for 2 CPUs and 4 GB RAM, it takes 90ish s.

Anyways the crux is that with reduced resources and `num_tasks = 10`, the test takes higher times. With `num_tasks = 3` it takes much less time (makes sense).

Reply:
```suggestion
    num_tasks = 3 if sys.platform == "win32" else 10
```